osu.Game.Tests/Skins/LegacySkinDecoderTest.cs

```diff
@@ -15,14 +15,24 @@ public class LegacySkinDecoderTest
     {
         [TestCase(true)]
         [TestCase(false)]
-        public void TestDecodeSkinColours(bool hasColours)
+        [TestCase(false, false)]
+        public void TestDecodeSkinColours(bool hasColours, bool canFallback = true)
```

Comment:
Please split out into multiple tests (at least the `canFallback` part). Each method should test at most one case.

Reply:
Does this mean to not always use `TestCase(val, val)` in future changes?

Reply:
In general, yes, they should be avoided. The typical use case is having expected values and such, e.g.:

`TestMethod(bool arg1, bool expectedResult)`

or similar. Having them as args that affect the test case increases the human parsing complexity a lot.